actionview/lib/action_view/template/renderable.rb

```diff
@@ -4,17 +4,31 @@ module ActionView
   class Template
     # = Action View Renderable Template for objects that respond to #render_in
     class Renderable # :nodoc:
-      def initialize(renderable)
+      def initialize(renderable, &block)
         @renderable = renderable
+        @block = block
       end
 
       def identifier
         @renderable.class.name
       end
 
-      def render(context, *args)
-        @renderable.render_in(context)
-      rescue NoMethodError
+      def render(context, locals)
+        options =
+          if @renderable.method(:render_in).arity == 1
+            ActionView.deprecator.warn <<~WARN
+              Action View support for #render_in without options is deprecated.
+
+              Change #render_in to accept keyword arguments.
+            WARN
+
+            {}
+          else
+            { locals: locals }
+          end
+
+        @renderable.render_in(context, **options, &@block)
+      rescue NameError
```

Comment:
@jonathanhefner since you helped me workshop the `NameError` handling below specific to `NameError` instances that stem from `#render_in` and **only** `#render_in`, I'm curious if a similar approach would be safe for `ArgumentError` handling:

```suggestion
      def render(context, locals)
        @renderable.render_in(context, locals: locals, &@block)
      rescue ArgumentError
        if @renderable.method(:render_in).arity == 1
          ActionView.deprecator.warn <<~WARN
            Action View support for #render_in without options is deprecated.

            Change #render_in to accept keyword arguments.
          WARN

          @renderable.render_in(context, &@block)
        else
          raise
        end
      rescue NameError
```

The motivation here is to optimize adherents to the interface to avoid slowing the happy path down with conditionals. Does that feel like a good trade-off?